staging/src/k8s.io/legacy-cloud-providers/aws/aws_routes.go

```diff
@@ -127,13 +127,40 @@ func (c *Cloud) ListRoutes(ctx context.Context, clusterName string) ([]*cloudpro
 	return routes, nil
 }
 
-// Sets the instance attribute "source-dest-check" to the specified value
-func (c *Cloud) configureInstanceSourceDestCheck(instanceID string, sourceDestCheck bool) error {
-	request := &ec2.ModifyInstanceAttributeInput{}
-	request.InstanceId = aws.String(instanceID)
-	request.SourceDestCheck = &ec2.AttributeBooleanValue{Value: aws.Bool(sourceDestCheck)}
-
-	_, err := c.ec2.ModifyInstanceAttribute(request)
+// Sets the instance's primary network interface attribute "source-dest-check" to the specified value
+func (c *Cloud) configureInstanceSourceDestCheck(instance *ec2.Instance, sourceDestCheck bool) error {
+	instanceID := aws.StringValue(instance.InstanceId)
+	srcDstCheck := &ec2.AttributeBooleanValue{Value: aws.Bool(sourceDestCheck)}
+
+	var err error
+	switch len(instance.NetworkInterfaces) {
```

Comment:
I think it's a little awkward to use switch to checking lenth of a slice. Just check using if statements?

Reply:
I prefer `switch` statement if there are more than 2 cases.

Reply:
While I agree with this in general, each case here is returning early so the code would be much easier to read as:
```
if len(instance.NetworkInterfaces) == 0 {
   return ...
}

if len(instance.NetworkInterfaces) == 1 {
   return ..
}

// when > 1 network interfaces.
```

Reply:
Updated as you said, PTAL.